examples/voting/src/lib.rs

```diff
@@ -0,0 +1,415 @@
+    };
+
+    Ok(result)
+}
+
+/// Get the election information.
+#[receive(contract = "voting", name = "view", return_value = "VotingView")]
+fn view<S: HasStateApi>(
+    _ctx: &impl HasReceiveContext,
+    host: &impl HasHost<State<S>, StateApiType = S>,
+) -> ReceiveResult<VotingView> {
+    // Get information from the state.
+    let description = host.state().description.clone();
+    let end_time = host.state().end_time;
+    let num_options = host.state().num_options;
+    let options = host.state().options.clone();
+
+    // Return the election information.
+    Ok(VotingView {
+        description,
+        end_time,
+        options,
+        num_options,
+    })
+}
+
+// Tests
+
+#[concordium_cfg_test]
+mod tests {
+    use super::*;
+    use concordium_std::test_infrastructure::*;
+
+    // Test accounts/addresses.
+    const ACC_0: AccountAddress = AccountAddress([0; 32]);
+    const ACC_1: AccountAddress = AccountAddress([1; 32]);
+    const ADDR_ACC_0: Address = Address::Account(ACC_0);
+    const ADDR_ACC_1: Address = Address::Account(ACC_1);
+
+    // Set up the test host.
+    fn make_test_host(
+        options: Vec<String>,
+        num_options: u32,
+        end_time: Timestamp,
+    ) -> TestHost<State<TestStateApi>> {
+        let mut state_builder = TestStateBuilder::new();
+        let state = State {
+            description: "Test description".into(),
+            ballots: state_builder.new_map(),
+            tally: state_builder.new_map(),
+            end_time,
+            options,
+            num_options,
+        };
+        TestHost::new(state, state_builder)
+    }
+
+    /// Test that an account cannot vote if it is past the end_time of the
+    /// election.
+    #[concordium_test]
+    fn test_vote_after_finish_time() {
+        // Set up the context.
+        let end_time = Timestamp::from_timestamp_millis(100);
+        let current_time = Timestamp::from_timestamp_millis(200);
+        let mut ctx = TestReceiveContext::empty();
+        ctx.set_metadata_slot_time(current_time);
+
+        // Set up logger.
+        let mut logger = TestLogger::init();
+
+        // Set up the test host.
+        let mut host = make_test_host(Vec::new(), 0, end_time);
+
+        // Call the contract function.
+        let res = vote(&ctx, &mut host, &mut logger);
+
+        // Check that error is thrown because the election is finished already.
+        claim_eq!(res, Err(VotingError::VotingFinished));
+    }
+
+    /// Test that voting fails if the voting index is out of range.
+    #[concordium_test]
+    fn test_vote_with_invalid_index() {
+        // Set up the context.
+        let end_time = Timestamp::from_timestamp_millis(100);
+        let current_time = Timestamp::from_timestamp_millis(0);
+        let mut ctx = TestReceiveContext::empty();
+        ctx.set_metadata_slot_time(current_time);
+        ctx.set_sender(ADDR_ACC_0);
+
+        // Set up the parameter.
+        let vote_parameter = to_bytes(&4);
```

Comment:
We could check the border case here instead, as that is easier to mess up (`<` vs `<=`).
```suggestion
        let vote_parameter = to_bytes(&2);
```